sunpy/net/dataretriever/client.py

```diff
@@ -225,7 +107,34 @@ def _can_handle_query(cls, *query):
         `sunpy.net.fido_factory.UnifiedDownloaderFactory`
         class uses to dispatch queries to this Client.
         """
-        raise NotImplementedError
+        from sunpy.net import attrs as a
+
+        required = {a.Time, a.Instrument}
```

Comment:
This needs to be a class attr?

Reply:
For most clients, `Instrument` and `Time` are required attrs so the method is same for all clients.
If any client deviates, it will be implemented in the  subclass.

Reply:
yeah but they shouldn't need to override this method to set this, they should just be able to set a class attr?

Reply:
This is still in the method and not a class attr?

Reply:
yes, since it is by default `instrument` and `Time`. If any client want to overwrite it, then it can be defined as class attributes for that clients, and the generic `_can_handle_query` checks if they are defined as class attributes or not. If not, it will follow the default rule.

Reply:
But why not just set required on this class, so it's always defined and then use it consistently in this method rather than doing `hasattr`. I would also think it would be a good idea to define `optional = set()` as a class attr as well, so these are always defined and we don't have to check if they are there.

Reply:
Done